shared/fs/banner/public-reminder.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import openUrl from '../../util/open-url'
+import * as FsGen from '../../actions/fs-gen'
+import * as Container from '../../util/container'
+
+type OwnProps = {
+  path: Types.Path
+}
+
+type Props = {
+  hidden: boolean
+  onClose: () => void
+  onLoad: () => void
+  show: boolean
+  url: string
+}
+const PublicBanner = (props: Props) => {
+  if (props.show) {
+    return (
+      props.hidden && (
+        <Kb.Banner
+          color="yellow"
+          text="Everything you upload in here can be viewed by everyone at "
+          actions={[
+            {
+              onClick: () => openUrl(props.url),
+              title: props.url + '.',
+            },
+          ]}
+          onClose={props.onClose}
+        />
+      )
+    )
+  } else {
+    props.onLoad()
```

Comment:
Yeah, this needs to be in lifecycles or `useEffect` rather than `render()`.

An alternative is to have the connector pass in simpler actions (i.e. `onSetLastClosed` and `clearLastClosed`), and have this component whether we should set or clear or do nothing.